File: text_ code/chap05/rwxlsx.py
# -*- coding: utf-8 -*-
"""
Created on Wed Oct 30 09:21:44 2019

@author: Sim
"""

import pandas as pd

data = pd.read_excel (r'K:\HWP\books\PR4DS\height.xlsx', header=None, 
                      names=['height','gender']) 
df1 = pd.DataFrame(data)
print(df1)


data = pd.read_excel ('http://jupiter.hallym.ac.kr/ftpdata/data/bmi.xlsx',
                     index_col=2) 
df2 = pd.DataFrame(data) #, columns= ['Product','Price'])
print(df2)

cars = {'make': ['Hyundai','Kia','Ford','Chevrolet'],
        'model': ['Sonata', 'K5', 'Taurus', 'Impala'],
        'price':  [3200,3100,3500,3700] }
df3 = pd.DataFrame(cars)
export_excel = df3.to_excel (r'K:\HWP\books\PR4DS\cars.xlsx', 
  index = None, header=True) #Don't forget to add '.xlsx' at the end of the path

'''
'''

File: text_ code/chap05/rwxlsx.r
# install.packages('readxl')
library(readxl)
# 전체 읽기
read_excel(path="k:/hwp/books/PR4DS/bmi.xlsx", sheet=1, 
      col_names=c("height", "weight", "year", "religion", "gender", "marriage"))
# A1:C10 영역만 읽기
read_excel(path="k:/hwp/books/PR4DS/bmi.xlsx", sheet=1, 
          range=("A1:C10"), col_names=c("height", "weight", "year"))
# 몸무게와 성별만 읽기
read_excel(path="k:/hwp/books/PR4DS/bmi.xlsx", sheet=1, 
      col_names=c("height", "weight", "year", "religion", "gender", "marriage"),
      col_types=c("skip", "numeric", "skip", "skip", "text", "skip"))

####################
# install.packages("xlsx")
library(xlsx)
read.xlsx(file="k:/hwp/books/PR4DS/bmi.xlsx", sheetIndex=1)
 
name <- c("강대성", "한준호", "김종욱", "박상호", "김소현")
number <- c(87, 73, 53, 65, 69)
df <- data.frame(name, number)
 
write.xlsx(df, "k:/hwp/books/PR4DS/scorebyR.xlsx", 
    sheetName="data", row.names=FALSE)


